### ISSUE TITLE
button press Taptic feedback

### ISSUE THREAD

**Issue by syto203** [iNDS-Team/iNDS#13]
add taptic engine support for button presses

**Comment by FrederickGeek8** [comment]
Is this not already implemented in the system? There should be an option in `Emulator Settings` labeled `Vibration`. Is that what you are looking for?

**Comment by syto203** [comment]
no. Starting from iPhone 6S Apple added a Taptic Engine alongside the vibrator.

**Comment by FrederickGeek8** [comment]
I just checked out that emulator and they are implementing a method that is in line with what I was reading with Apple documentation. I had actually started to implement this before I realized that we already had a Vibration feature.

**Comment by FrederickGeek8** [comment]
**_TLDR: I would really like to hear from everybody. Which sounds like a better haptic feedback? Do you want something more forceful and that feels like a button press? Or more subtle, something that might not bug you after hours of usage? You can react to this post with a 🎉if you want button-like feedback, or a ❤️ if you want softer feedback, or just comment or post in Gitter._** 

There are really two options as to what we want to use for our vibration settings. They are similar, but definitely feel different. 

**Impact vs. Selection**
Apple defines two non-notification tapic engine events, one which they describe for navigating a selection (like a scroll wheel) or one for impacts (like two objects in a view colliding). The description given by Apple for each leaves a lot to be desired, for example, they specify that the selection feedback should be use not for final selection, but when navigating discrete values.

The *Provenance* emulator uses the *Selection Changed* feedback mechanism, however, I'm not completely sure I agree with that decision. I find that the Selection Changed feedback is too subtle and doesn't give that satifing click that I would desire in haptic feedback. Additionaly, given Apple's *somewhat* sketchy documentation, it is unclear to be if that would be its intended purpose (I do realize that Apple probably would discourage any sort of haptic feedback associated with buttons).

Below I have placed the descriptions of each as per Apple documentation. [Apple has a nice interactive website](https://developer.apple.com/design/human-interface-guidelines/ios/user-interaction/feedback/#haptics) to audibly get a feel for each, if you cannot test out any code. On that note, I have [code up on FrederickGeek8/iNDS/tree/add-taptic](https://github.com/FrederickGeek8/iNDS/tree/add-taptic) which you can use to try out the feel of both. You can go to [iNDS/iNDSEmulatorViewController.mm (line 1001)](https://github.com/FrederickGeek8/iNDS/blob/add-taptic/iNDS/iNDSEmulatorViewController.mm#L1001) and comment-swap out the two different taptic-feedback mechanics and see which one you like better. 

**I would really like to hear from everybody. Which sounds like a better haptic feedback? Do you want something more forceful, that feels like a button press? Or more subtle, something that might not bug you after hours of usage? You can react to this post with a 🎉if you want button-like feedback, or a ❤️ if you want softer feedback.**

*Light Impact Description*

> Provides a physical metaphor that complements the visual experience. For example, the user might feel a thud when a view slides into place or two objects collide.

*Selection Changed Description*

> Indicates that the selection is actively changing. For example, the user feels light taps while scrolling a picker wheel. This feedback is intended for communicating movement through a series of discrete values, not making or confirming a selection.
